Plugin.pm

```diff
@@ -954,31 +959,74 @@ sub QobuzGetTracks {
 					image => $formattedTrack->{image},
 					tracks => []
 				} unless $works->{$workId};
-
+				
+				if ($workId ne $lastwork) {  # create a new work heading
+#					if ($i && $lastwork eq "") {  # create a separator line after previous single track#
+#						push @$items,{
+#							name  => "--------",
+#							type  => 'text'
+#						};
+#					}
+					push @$items,{
+						name  => $work,
+						type  => 'text'
+					};
+						
+					$lastwork = $workId;
+				}
+				
 				push @{$works->{$workId}->{tracks}}, $formattedTrack;
-			}
+				if (scalar @{$works->{$workId}->{tracks}} > 1) {
+					$worksfound = 1;   # we found a work with more than one track
+				}
+			} elsif ($lastwork ne "") {  # create a separator line for tracks without a work
+						push @$items,{
+							name  => "--------",
+							type  => 'text'
+						};
+						
+					$lastwork = "";
+			}	
```

Comment:
Please indent only one level.

Reply:
And you could give an m-dash "—" a try to have a line without spaces

————————